From 64aa033a061c43fc15c711f2490ae41d23b868c3 Mon Sep 17 00:00:00 2001
From: Fabio Berton <fabio.berton@ossystems.com.br>
Date: Thu, 17 Nov 2016 09:44:42 -0200
Subject: [PATCH 1/2] Fix compiler_state_t.ai usage when INET6 is not defined
Organization: O.S. Systems Software LTDA.

Fix error:

/
| ../libpcap-1.8.1/gencode.c: In function 'pcap_compile':
| ../libpcap-1.8.1/gencode.c:693:8: error: 'compiler_state_t
| {aka struct _compiler_state}' has no member named 'ai'
|   cstate.ai = NULL;
\

Upstream-Status: Submitted [1]

[1] https://github.com/the-tcpdump-group/libpcap/pull/541

Signed-off-by: Fabio Berton <fabio.berton@ossystems.com.br>
---
 gencode.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/gencode.c b/gencode.c
index a887f27..e103c70 100644
--- a/gencode.c
+++ b/gencode.c
@@ -690,7 +690,9 @@ pcap_compile(pcap_t *p, struct bpf_program *program,
 	}
 	initchunks(&cstate);
 	cstate.no_optimize = 0;
+#ifdef INET6
 	cstate.ai = NULL;
+#endif
 	cstate.ic.root = NULL;
 	cstate.ic.cur_mark = 0;
 	cstate.bpf_pcap = p;
-- 
2.1.4

